packages/sparrowql/src/build.ts

```diff
@@ -10,321 +11,322 @@ export type Relation = {
-
-        const entries = Object.entries(sort);
-        if (entries.length === 0) {
-            sort = undefined;
-            return;
-        }
-
-        if (entries.every(entry => joined.includes(getNameCollection(entry[0])))) {
-            steps.push({
-                sort: entries.reduce(
-                    (object, entry) => Object.assign(object, {[relative(entry[0], false)]: entry[1]}),
-                    {}
-                )
-            });
-            sort = undefined;
-        }
+      });
+      sort = undefined;
     }
+  }
 }
 
-const translateOperators = {
-    group: (start: string, step: Step) => [{$group: step.group}],
-    limit: (start: string, step: Step) => [{$limit: step.limit}],
-    match: (start: string, step: Step) => [{$match: step.match}],
-    projection: (start: string, step: Step) => [{$project: step.projection}],
-    relation: (start: string, step: Step) => [
-        {
-            $lookup: {
-                as: getNameRelative(start, step.relation!.to, false),
-                foreignField: step.relation!.foreign,
-                from: step.relation!.toAlias,
-                localField: step.relation!.local
-            }
-        },
-        {
-            $unwind: {
-                path: getNameRelative(start, step.relation!.to, true),
-                preserveNullAndEmptyArrays: true
-            }
-        }
-    ],
-    skip: (start: string, step: Step) => [{$skip: step.skip}],
-    sort: (start: string, step: Step) => [{$sort: step.sort}]
+const translateOperators: Record<'group' | 'limit' | 'match' | 'projection' | 'relation' | 'skip' | 'sort', <T>(start: string, step: Step<T>) => any> = {
+  group: <T = {}>(start: string, step: Step<T>) => [{$group: step.group}],
+  limit: <T = {}>(start: string, step: Step<T>) => [{$limit: step.limit}],
+  match: <T = {}>(start: string, step: Step<T>) => [{$match: step.match}],
+  projection: <T = {}>(start: string, step: Step<T>) => [{$project: step.projection}],
+  relation: <T = {}>(start: string, step: Step<T>) => [
+    {
+      $lookup: {
+        as: getNameRelative(start, step.relation!.to, false),
+        foreignField: step.relation!.foreign,
+        from: step.relation!.toAlias,
+        localField: step.relation!.local
+      }
+    },
+    {
+      $unwind: {
+        path: getNameRelative(start, step.relation!.to, true),
+        preserveNullAndEmptyArrays: true
+      }
+    }
+  ],
+  skip: <T = {}>(start: string, step: Step<T>) => [{$skip: step.skip}],
+  sort: <T = {}>(start: string, step: Step<T>) => [{$sort: step.sort}]
 };
 
-export function translate(start: string, steps: Step[]) {
-    const pipeline = [];
-
-    for (const step of steps) {
-        const operators = Object.keys(step) as (keyof Step)[];
-        if (operators.length !== 1) {
-            throw new Error(`Invalid step: ${JSON.stringify(step)}`);
-        }
+export function translate <T = {}>(start: string, steps: Step<T>[]) {
+  const pipeline = [];
```

Comment:
I think it's `any` now, therefore here's my suggestion:
```suggestion
  const pipeline: object[] = [];
```